integration-tests/ahm/src/mock.rs

```diff
@@ -90,3 +98,89 @@ pub fn enqueue_dmp(msgs: Vec<InboundDownwardMessage>) {
 		);
 	}
 }
+
+// Migrates the pallet out of the Relay Chain and returns the corresponding Payload.
+//
+// Sends DMP messages with pallet migration data from relay chain to asset hub. The output includes
+// both the DMP messages sent from the relay chain to asset hub, which will be used to perform the
+// migration, and the relay chain payload, which will be used to check the correctness of the
+// migration process.
+pub fn rc_migrate<RcMigrator: RcPalletMigrationChecks>(
```

Comment:
@re-gius whats the purpose for these new abstractions? `rc_migrate` and `ah_migrate`?